prow/repoowners/repoowners.go

```diff
@@ -806,7 +808,7 @@ func (o *RepoOwners) entriesForFile(path string, people map[string]map[*regexp.R
 		}
 		for re, s := range people[d] {
 			if re == nil || re.MatchString(relative) {
-				out.Insert(s.List()...)
+				out.Insert(layerID, s.List()...)
```

Comment:
@BenTheElder the real trick is here, where we no longer mix owners from different levels